content/blog/20220331_fluentd-v1.14.6-has-been-released.md

```diff
@@ -0,0 +1,146 @@
+# Fluentd v1.14.6 has been released
+
+Hi users!
+
+We have released v1.14.6. ChangeLog is [here](https://github.com/fluent/fluentd/blob/master/CHANGELOG.md#release-v1146---20220331).
+
+This release is a maintenance release of v1.14 series.
+
+## Enhancement
+
+### Enable server plugins to specify socket-option `SO_LINGER`
+
+In this release, we add a new option `linger_timeout` to `server` plugin-helper
+so that we can specify `SO_LINGER` socket-option when using TCP or TLS server function of the helper.
+
+The helper has used `0` for `linger_timeout` internally before. This made the connection closing immediate,
+and especially on non-Windows, this made the server send RST rather than FIN on the closing.
+
+The main use of this option is to make the server send FIN on closing on non-Windows,
+by setting a positive value such as `1`.
+
+This new option is in `transport` section, and we can specify this for plugins
+using TCP or TLS server function of the helper, such as `in_tcp`, `in_http`,
+`in_syslog`, and `in_forward`. 
+
+```
+<source>
+  @type tcp
+  ...
+  <parse>
+    ...
+  </parse>
+  <transport tcp>
+    linger_timeout 1
+  </transport>
+</source>
+```
+
+Note: `in_forward` plugin had `linger_timeout` option before, but the option is now deprecated and unified to `transport` section.
+
+See [#3644](https://github.com/fluent/fluentd/pull/3644) for more information.
+
+### Add `--umask` to `fluentd` command line parameter
+
+When we launch Fluentd with `--no-supervisor`, the umask value of the environment
+is not applied because Fluentd overwrites the value.
+
+In this release, We can use `--umask` option as a `fluentd` command line parameter
+to specify the umask value in the case of launching Fluentd with `--no-supervisor`.
+
+```console
+$ fluentd -c {config_path} --no-supervisor --umask 002
+```
+
+See [#3671](https://github.com/fluent/fluentd/pull/3671), [#3679](https://github.com/fluent/fluentd/pull/3679) for more information.
+
+## Bug Fixes
+
+### Fix wrong calculation of retry count and interval
+
+There were mistakes in the calculation about `exponential_backoff` retry_type.
+
+In this release, the calculation is fixed, and the count and the interval change as following.
+
+**Count**
+
+* Before: At most `retry_max_times + 1` retries were triggered.
+* This Release: At most `retry_max_times` retries are triggered.
+
+**Interval**
+
+* Before: By default, n-th retry was triggered after 2^(n-1) seconds elapsed since the **first** time.
```

Comment:
Doesn't it mean that the time elapsed since the first flush is eventually `2^(n-1)`?

first flush: 0s
1st retry: 1s
2st retry: 2s
3st retry: 4s
...

Reply:
Ah, I see.